src/connectors/sink.rs

```diff
@@ -0,0 +1,186 @@
+// Copyright 2021, The Tremor Team
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use crate::codec::Codec;
+use crate::errors::Result;
+use crate::pipeline;
+use crate::postprocessor::Postprocessors;
+use crate::url::TremorUrl;
+use async_channel::{bounded, Receiver, Sender};
+use beef::Cow;
+use hashbrown::HashMap;
+use tremor_pipeline::Event;
+
+pub enum SinkReply {
+    Contraflow(Event),
+}
+
+/// Result for a sink function that may provide insights or response.
+///
+/// It can return None or Some(vec![]) if no insights/response were generated.
+///
+/// An insight is a contraflowevent containing control information for the runtime like
+/// circuit breaker events, guaranteed delivery events, etc.
+///
+/// A response is an event generated from the sink delivery.
+pub(crate) type ResultVec = Result<Option<Vec<SinkReply>>>;
```

Comment:
I know this came to originated somewhere else, but I wonder would it make sense to simplify this to `Result<Vec<SinkReply>>` or is there a difference between an empty Vec and a None?

Reply:
good point